google/cloud/aiplatform/datasets.py

```diff
@@ -15,6 +15,358 @@
+                Points to a YAML file stored on Google Cloud
+                Storage describing the import format. Validation will be
+                done against the schema. The schema is defined as an
+                `OpenAPI 3.0.2 Schema
+                Object <https://tinyurl.com/y538mdwt>`__.
+            project: Optional[str]=None,
+                Project to upload this model to. Overrides project set in
+                aiplatform.init.
+            location: Optional[str]=None,
+                Location to upload this model to. Overrides location set in
+                aiplatform.init.
+            credentials: Optional[auth_credentials.Credentials]=None,
+                Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+        Returns:
+            dataset (Dataset):
+                Instantiated representation of the managed dataset resource.
+        """
+
+        # Validate that source and import schema are passed together or not at all
+        if bool(source) ^ bool(import_schema_uri):
+            raise ValueError(
+                "Please provide both source and import_schema_uri to import data"
+            )
+
+        api_client = cls._instantiate_client(location=location, credentials=credentials)
+
+        create_dataset_lro = cls._create(
+            display_name=display_name,
+            parent=initializer.global_config.common_location_path(
+                project=project, location=location
+            ),
+            metadata_schema_uri=metadata_schema_uri,
+            request_metadata=metadata,
+            labels=labels,
+            api_client=api_client,
+        )
+
+        created_dataset = create_dataset_lro.result()
+
+        dataset_obj = cls(
+            dataset_name=created_dataset.name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        # If an import source was not provided, return empty created Dataset.
+        if not source:
+            return dataset_obj
+
+        return dataset_obj.import_data(
+            gcs_source=source,
+            import_schema_uri=import_schema_uri,
+            data_items_labels=data_items_labels,
+        )
+
+    @classmethod
+    def _create(
+        cls,
+        api_client: DatasetServiceClient,
+        parent: str,
+        display_name: str,
+        metadata_schema_uri: str,
+        labels: Optional[Dict] = {},
+        request_metadata: Sequence[Tuple[str, str]] = (),
+    ) -> operation.Operation:
+
+        gapic_dataset = GapicDataset(
+            display_name=display_name,
+            metadata_schema_uri=metadata_schema_uri,
+            labels=labels,
+        )
+
+        return api_client.create_dataset(
+            parent=parent, dataset=gapic_dataset, metadata=request_metadata
+        )
+
+    def _import(
+        self,
+        source: Union[str, List[str]],
```

Comment:
I think for source we should accept Sequence over List, throughout the code, because it matches the type or GcsSource.uris.

For type checking, to determine whether to envelope the string in a list, check for a string instead of a list.

Reply:
Replacing `Union[str, List[str]]` with `Sequence[str]`